tools/permissions.ts

```diff
@@ -0,0 +1,298 @@
+import { z } from "zod";
+import { zodToJsonSchema } from "zod-to-json-schema";
+import { jfrogRequest } from "../common/utils.js";
+
+// Schema definitions
+const ResourceTargetSchema = z.object({
+  include_patterns: z.array(z.string()),
+  exclude_patterns: z.array(z.string())
+});
+
+const ResourceActionsSchema = z.object({
+  users: z.record(z.array(z.enum([
+    'READ',
+    'WRITE',
+    'ANNOTATE',
+    'DELETE',
+    'DISTRIBUTE',
+    'MANAGE'
+  ]))),
+  groups: z.record(z.array(z.enum([
+    'READ',
+    'WRITE',
+    'ANNOTATE',
+    'DELETE',
+    'DISTRIBUTE',
+    'MANAGE'
+  ]))).optional()
+});
+
+const ArtifactResourceSchema = z.object({
+  actions: ResourceActionsSchema,
+  targets: z.record(ResourceTargetSchema)
+});
+
+const ReleaseBundleResourceSchema = z.object({
+  actions: ResourceActionsSchema,
+  targets: z.record(ResourceTargetSchema)
+});
+
+const BuildResourceSchema = z.object({
+  actions: ResourceActionsSchema,
+  targets: z.record(ResourceTargetSchema)
+});
+
+const PermissionTargetSchema = z.object({
+  name: z.string(),
+  resources: z.object({
+    artifact: ArtifactResourceSchema.optional(),
+    release_bundle: ReleaseBundleResourceSchema.optional(),
+    build: BuildResourceSchema.optional()
+  }),
+  created_by: z.string().optional(),
+  modified_by: z.string().optional()
+});
+
+const PermissionListSchema = z.object({
```

Comment:
Could you please move all the schemas into a dedicated schema folder for better organization? Also, since permissions are part of the access tools logic, they can be placed under the access tools file.